### ISSUE TITLE
No such file or directory

### ISSUE THREAD

**Issue by heropoo** [vn971/rua#58]
A error
 ```
thread 'main' panicked at 'Failed to find config dir for locking: Os { code: 2, kind: NotFound, message: "No such file or directory" }', src/libcore/result.rs:999:5
note: run with `RUST_BACKTRACE=1` environment variable to display a backtrace.
```
return when i run `rua search some-pkg-name`.

So i `mkdir -p ~/.config/rua`, the error is not display

**Comment by vn971** [comment]
@heropoo thanks for reporting! The problem is that I switch to not creating system directories for commands that do not require them. This requirement, on the other hand, is not controlled by the type system, and it's hard to miss (as it was in this issue).

In the commit that closes this issue, I started always creating system directories. Even for commands like `search`, which doesn't need _most_ of those things.

**Comment by vn971** [comment]
EDIT ^

**Comment by vn971** [comment]
.. And quite sadly I made another mistake regarding escaping yesterday. (Escaping should've been just for beautiful visual printing, but leaked elsewhere). Noticed and fixed now in release 0.14.9 b02da858873b0b0c458acf4713a8db178e9e0337
